### ISSUE TITLE
RewardCriterion mask

### ISSUE THREAD

**Issue by fawazsammani** [ruotianluo/self-critical.pytorch#137]
Hi Ruotian,
Thanks for your code!
I'd like to ask: For the RewardCriterion, in this line:
`mask = to_contiguous(torch.cat([mask.new(mask.size(0), 1).fill_(1), mask[:, :-1]], 1)).view(-1)`
Why are you concatenating the `mask[:, :-1]` with 1s at the beginning `mask.size(0), 1).fill_(1)`? Since the mask comes from original `seq` (which will be multiplied with the logprobs of the same shape as seq), why are you changing it? Perhaps you start with 1s since there is always a word at the begining, but then why are you ignoring the last row in the mask?

**Comment by fawazsammani** [comment]
I found the answer in this issue: #102 
Sorry for not searching!
